Pawn.lua

```diff
@@ -780,6 +786,25 @@ function PawnLinkWranglerOnTooltip(Tooltip, ItemLink)
 	PawnAttachIconToTooltip(Tooltip, false, ItemLink)
 end
 
+-- ArkInventory Rule
+function ArkInventoryPawnUpgradeRule( ... )
+    local fn = "PAWNUPGRADE" -- Rule name for errors
+    if not PawnIsInitialized then VgerCore.Fail("Can't check to see if items are upgrades until Pawn is initialized") return end
+
+    -- Verify that the item string information is loaded and not nil and that it is a valid item before continuing
+    if not ArkInventoryRules.Object.h or ArkInventoryRules.Object.class ~= "item" then return false end
+
+    -- Parse the incoming item and retrieve the data
+    local ac = select( '#', ... )
+    local info = ArkInventory.ObjectInfoArray( ArkInventoryRules.Object.h )
+
+    -- Extract the itemLink from the ArkInventory info object
+    local itemLink = info.info[2]
+
+    -- Use the same logic for determining whether or not an arrow should be shown, for consistency
+    return PawnShouldItemLinkHaveUpgradeArrow(itemLink, true)
```

Comment:
This will return `nil` if WoW hasn't downloaded item stats yet or Pawn otherwise can't determine if it's an upgrade or not. If there similarly isn't a way to flag an upgrade rule as "I don't know", maybe `ArkInventoryPawnUpgradeRule` should return `true` when `PawnShouldItemLinkHaveUpgradeArrow` returns `nil` just so there isn't a chance to vendor an item that Pawn isn't sure about yet?

Reply:
(Same comment on the other `return`s in this function)

Reply:
Agreed. I'll send a new revision with that. I was also considering adding different implementations of the rule:

|given|when|then
|---|---|---|
|PawnShouldItemLinkHaveUpgradeArrow returns true| pawnupgrade() rule used| return true|
|PawnShouldItemLinkHaveUpgradeArrow returns nil| pawnupgrade() rule used| return false|
|PawnShouldItemLinkHaveUpgradeArrow returns false| pawnupgrade() rule used| return false|
|PawnShouldItemLinkHaveUpgradeArrow returns true| pawnnotupgrade() rule used| return false|
|PawnShouldItemLinkHaveUpgradeArrow returns nil| pawnnotupgrade() rule used| return false|
|PawnShouldItemLinkHaveUpgradeArrow returns false| pawnnotupgrade() rule used| return true|


This would convey the certainty from pawn one way or another, and if users really wanted to find things that pawn is unsure about, this gives them the tools do so `((not pawnupgrade()) and (not pawnnotupgrade())`. (Side question: is pawndowngrade() better?)

Thoughts?

Reply:
I'd stick with "not upgrade" versus "downgrade"—I do want to include options *someday* for showing sidegrades and downgrades, and they'll probably use subtly different logic than this, so that might be confusing.

Reply:
Sounds good to me. Will send an update to the PR shortly.

Reply:
So for the second function, what you're looking for is kind of a "`PawnIsItemJunk`" that helps you find items that *can* currently be equipped (including meeting level requirements), but *aren't* an upgrade. Does that sound right?

If you want to copy and paste `PawnShouldItemLinkHaveUpgradeArrow` and make those changes that's fine with me. You can remove the parts about `DebugBagArrows` and Legion relics so the new function wouldn't be that big. This stuff is probably getting pretty ripe for refactoring, but I don't think I want to tackle that right now.